### PR TITLE
Initial commit to add the base pipeline component.

### PR DESCRIPTION
**Pull request by msjonathan**
Initial commit

### CODE REVIEW

**Comment by tomkerkhove** [comment]
I agree on most part but I'm not sure if we need the `IsEnabled` property... What do you think @samneirinck and @GlennColpaert?

File: src/BizTalk.Extended.Pipelines.General/PipelineComponent.cs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+        /// </summary>
+        /// <param name="pipelineContext"></param>
+        /// <param name="message"></param>
+        /// <returns></returns>
+        public IBaseMessage Execute(IPipelineContext pipelineContext, IBaseMessage message)
+        {
+            Guard.NotNull(pipelineContext, "pipelineContext");
+            Guard.NotNull(message, "message");
+
+            IBaseMessage returnMessage = null;
```
Comment (tomkerkhove) [inline src/BizTalk.Extended.Pipelines.General/PipelineComponent.cs]: Maybe it's better to assign it directly to `message` and get rid of the `else` branch in the `try\catch`?

**Comment by msjonathan** [comment]
I kept the IsEnabled property because i thought it was a nice operations feature, an operator can disable when a component that is not working properly. 
Or in some cases when having a generic pipeline, somethimes you want to disable a component in a receive port / send port to have different behavior. 

But I can understand your thinking, maybe the basecomponent should be a naked component without any properties. Should we arrange a poll?

**Comment by samneirinck** [comment]
I think an IsEnabled property is a good idea. Typically, it's not something you write on your own if you develop a custom pipeline component, but it can come in very handy once you need it and figure out it's already in there, for free. 

The purpose is also very clear, enable or disable the component, so it won't cause any confusion. The default is also set to true, so it won't cause any issues if you just ignore it, plus it's shielded away in the base class so your ExecuteInternal is clean as well.

**Comment by tomkerkhove** [comment]
I completely agree, however I can understand that someone might expect a clean `PipelineComponent` where the overhead is shielded from me. I'd rather see the one with a property as a `ToggelablePipelineComponent` or whatsoever.

But if we decide to go with the property in the `PipelineComponent` I can live with that as well.

**Comment by tomkerkhove** [comment]
As agreed we will merge this request. If we notice that the property is not what people want we can tear it apart later on. Thanks @msjonathan!

**Pull request merged**